frontend/sozisel-app/src/components/PresenterSession/Agenda/Agenda.tsx

```diff
@@ -0,0 +1,111 @@
+import "./Agenda.scss";
+
+import { Paper, Typography } from "@material-ui/core";
+import React, { ReactElement, useEffect, useState } from "react";
+
+import { AgendaPoint } from "../../../model/Agenda";
+import TurnedInNotIcon from "@material-ui/icons/TurnedInNot";
+
+interface AgendaProps {
+  agendaEntries: AgendaPoint[];
+  estimatedTime: number;
+}
+
+interface AgendaEntryProps {
+  startMinute: number;
+  endMinute: number;
+  name: string;
+  idx: number;
+}
+
+export default function Agenda({
+  agendaEntries,
+  estimatedTime,
+}: AgendaProps): ReactElement {
+  const [counter, setCounter] = useState<number>(0);
+  const agendaStatus = (startMinute: number, endMinute: number) => {
+    if (counter < startMinute) return "";
+    if (
+      (counter >= startMinute && counter < endMinute) ||
+      startMinute === endMinute
+    )
+      return "during";
+    return "finished";
+  };
+
+  useEffect(() => {
+    if (estimatedTime && counter < estimatedTime) {
+      setTimeout(() => {
+        setCounter(counter + 1);
+      }, 60000);
+    }
+  }, [counter, estimatedTime]);
+
+  const AgendaEntry = ({
+    startMinute,
+    endMinute,
+    name,
+    idx,
+  }: AgendaEntryProps): ReactElement => {
+    return (
+      <>
+        <Typography
+          className={`agendaHeader ${agendaStatus(startMinute, endMinute)}`}
+        >
+          {idx + 1}. {name}
+        </Typography>
+        <Typography
+          className={`agendaHeader ${agendaStatus(startMinute, endMinute)}`}
+        >
+          {endMinute} min
+        </Typography>
+      </>
+    );
+  };
+
+  return (
+    <Paper elevation={2} className="Agenda">
+      <Typography className="header">
+        <TurnedInNotIcon className="icon" />
+        Agenda
+      </Typography>
+      {agendaEntries?.map((agendaEntry: AgendaPoint, idx: number) => (
+        <div key={`${idx}-${agendaEntry.name}`} className="agendaEntry">
+          {(() => {
```

Comment:
Change that into a separate function. And one question, why is it so complicated?

Reply:
And I have not idea what you've done here with that arrow function and then calling it, wrapping it in another set of parenthesis...

Reply:
It is so complicated because we have 2 extreme cases except for the normal case which we have to handle:
1. When we have the first agenda entry, by default it has to be the `ON_GOING` stage.
2. When we have the last agenda entry in the `ON_GOING` stage and the session is not over yet, this agenda entry must be in "main-color" until the session is over.

Reply:
So I don't see any more simple solution to handle these cases, but ok I will extract it to another function.